lib/ninject.rb

```diff
@@ -0,0 +1,12 @@
+class Array
+  def ninject(initial = nil, &proc)
+    each_with_index do |obj, index|
+      if index == 0
+        @memo = initial.nil? ? obj : proc.call(initial, obj)
```

Comment:
You shouldn't use an instance variable `@memo` as there is no need to persist the memo beyond this method call